aria-practices.html

```diff
@@ -5927,6 +5927,697 @@ <h5>Intentional Key Conflicts</h5>
+              <code>columnheader</code>,
+              <code>rowheader</code>
+            </td>
+          </tr>
+          <tr>
+            <td>
+              <code>listbox</code>
+            </td>
+            <td>
+              <code>option</code>
+            </td>
+          </tr>
+          <tr>
+            <td>
+              <code>tree</code>
+            </td>
+            <td>
+              <code>treeitem</code>
+            </td>
+          </tr>
+          <tr>
+            <td>
+              <code>tablist</code>
+            </td>
+            <td>
+              <code>tab</code>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+
+      <section id="communicating_widget_states_selecting_rows_and_cells">
+        <h4>Selecting rows and cells in <code>grid</code> and <code>treegrid</code></h4>
+
+        <p>A <code>gridcell</code> in a <code>grid</code> or <code>treegrid</code> supports the <code>aria-selected</code> state.
+          The widget could support selecting one or multiple cells.
+        </p>
+
+        <p>
+          TODO example.
+        </p>
+
+        <p>
+          The <code>row</code> role within a <code>grid</code>, <code>table</code> or <code>treegrid</code> element supports the <code>aria-selected</code> state.
+        </p>
+
+        <p>
+          See the <a href="examples/combobox/aria1.1pattern/grid-combo.html">ARIA 1.1 Combobox with Grid Popup Example</a>
+          of the <a href="#combobox">Combobox</a> design pattern.
+        </p>
+      </section>
+
+      <section id="communicating_widget_states_selecting_options">
+        <h4>Selecting options in listboxes and treeitems in trees</h4>
+
+        <p>
+          <code>aria-selected</code> is a required state for the <code>option</code> role in a <code>listbox</code> element.
+          When an item is selected, specify <code>aria-selected="true"</code>.
+          When an item is not selected, specify <code>aria-selected="false"</code>;
+          do not omit the attribute.
+        </p>
+
+        <p>
+          See the <a href="#combobox">Combobox</a> and the <a href="#Listbox">Listbox</a> design patterns for examples.
+        </p>
+
+        <p>
+          The <code>option</code> role and the subclass role <code>treeitem</code> support both <code>aria-selected</code> and <code>aria-checked</code>.
+          In most cases, using only <code>aria-selected</code> is the right choice.
+          Do not add both <code>aria-selected</code> and <code>aria-checked</code> "just in case",
+          since that would increase verbosity for assistive technology users and could cause unnecessary confusion.
+        </p>
+
+        <p>
+          In rare cases, a widget could support both selection and checkedness to represent orthogonal states,
+          although that could be complex and might be confusing for users.
+          For example, a <code>tree</code> widget to represent a changeable sequence of steps,
```

Comment:
Based on the discussion in the APG meeting today, we decided on removing this example, and adding guidance to indicate that a valid use of `aria-checked` over `aria-selected` is when you need to support a mixed selection state, but that both attributes should not be used at the same time.